{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -0,0 +1,125 @@
+# locations in a file called "locations.txt" in the same directory. You
+# can create a single location using the "-s" option. Using the single
+# location option will override an existing locations file.
+#
+# The application will look for the model spec and locations files based
+# on the python environment that is active and these files don't need
+# to be specified if the default names and location are used.
+# =============================================================================
+import re
+import click
+from pathlib import Path
+from jinja2 import Template
+from collections import namedtuple
+
+from typing import List
+
+
+PROJ_NAME='{{cookiecutter.package_name}}'
+MODEL_SPEC_DIR=Path(__file__).parent.parent / 'model_specifications'
+DEFAULT_TEMPLATE_FILE=MODEL_SPEC_DIR / 'model_spec.in'
+DEFAULT_LOCATIONS_FILE=MODEL_SPEC_DIR / 'locations.txt'
+Location = namedtuple('Location', ['loc_proper', 'loc_sanitized'])
+
+
+COMMENT_CHAR='#'
+REPLACE_WITH_UNDERSCORE="[- ,.&']"
+def sanitize(loc: str) -> (bool, Location):
+    loc_proper = loc.strip()
+    if loc.startswith(COMMENT_CHAR):
+        loc_proper = loc_sanitized = ''
+    else:
+        loc_sanitized = loc_proper
+        loc_sanitized = re.sub(REPLACE_WITH_UNDERSCORE, '_', loc_sanitized).lower()
+    return True if len(loc_proper) else False, Location(loc_proper, loc_sanitized)
+
+
+def get_sanitized_locations(single_loc : str, loc_file : Path) -> List[Location]:
+    """ Naming convention is to lowercase the location string, replace spaces with underscores,
+        and capitalize the first letter. No manipulation happens in the template code.
+    """
+    def helper(item):
+        valid, loc = sanitize(item)
+        if valid:
+            locs.append(loc)
+
+    locs = []
+    if len(single_loc):
+        helper(single_loc)
+    else:
+        with loc_file.open('r') as infile:
+            for line in infile:
+                helper(line)
+    return locs
+    
+    
+def args_pass(template: Path, locations_file: Path, single_location: str, output_dir: Path) -> bool:
+    error = ''
+    if not template.exists():
+        error = f'\nError: the template file {template} does not exist.'
+    elif not locations_file.exists():
+        error = f'\nError: the locations file {locations_file} does not exist. Create locations.txt ' \
+                'or use the "-s" switch to specify a single location.'
+    elif not len(single_location) and not locations_file.stat().st_size:
+        error = f'\nError: the locations file {locations_file} is empty. Add locations ' \
+                'to this file, or, use the "-s" switch to specify a single location.'
+    elif not output_dir.exists():
+        error = f'\nError: the output director {output_dir} does not exist.'
+
+    if len(error):
+        print(error)
+
+    return True if 0 == len(error) else False
+
+
+@click.command()
+@click.option('-l', '--locations_file',
+                default=DEFAULT_LOCATIONS_FILE,
```

Comment:
I think we could use fewer global variables and make things clearer.  E.g. just set the default here to `MODEL_SPEC_DIR / 'locations.txt'.  Similar for templates.